torch.ts/src/tensor.ts

```
@@ -0,0 +1,649 @@
+
+class TensorBinary implements TensorI {
+  l: TensorI;
+  r: TensorI;
+  op: (a: number, b: number) => number;
+
+  constructor(
+    l: TensorI,
+    r: TensorI,
+    op: (a: number, b: number) => number,
+  ) {
+    if (l.shape.length !== l.shape.length || !l.shape.every((dim, i) => dim === r.shape[i]))
```

Biome (2.1.2)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)